### PR TITLE
MONIT-34006: Remove Proxy Validations and limits when converged csp tenant setting is set

### PR DESCRIPTION
**Pull request by deosu**
- Add a new flag `enableHyperlogsConvergedCsp` to indicate if the customer is on CSP and tenants are converged (the same) btw WF and VRLIC.
- Skip `hostLengthLimit` to check for logs if the above tag is set to true.

### CODE REVIEW

File: java-lib/src/main/java/com/wavefront/api/agent/AgentConfiguration.java [hunk 1]
```diff
@@ -107,6 +107,12 @@ public class AgentConfiguration {
    */
   private String logServerToken;
 
+  /**
+   * This flag indicates if the customer is on CSP and tenants are converged (the same)
+   * btw WF and VRLIC
+   */
+  private boolean enableHyperlogsConvergedCsp = false;
```
[the remark this answers is not in the document]
Comment (deosu) [reply java-lib/src/main/java/com/wavefront/api/agent/AgentConfiguration.java]: @sudeep-tobs can you please take a look at it again?

**Review by sudeep-tobs — APPROVED** [review]
LGTM

File: java-lib/src/main/java/com/wavefront/api/agent/ValidationConfiguration.java [hunk 2]
```diff
@@ -212,6 +218,14 @@ public ValidationConfiguration setLogAnnotationsCountLimit(int value) {
     return this;
   }
 
+  public boolean isEnableHyperlogsConvergedCsp() {
+    return enableHyperlogsConvergedCsp;
+  }
+
+  public void setEnableHyperlogsConvergedCsp(boolean enableHyperlogsConvergedCsp) {
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/api/agent/ValidationConfiguration.java:225]: could we change return type to `ValidationConfiguration` for consistency?

File: java-lib/src/main/java/com/wavefront/api/agent/ValidationConfiguration.java [hunk 3]
```diff
@@ -212,6 +218,14 @@ public ValidationConfiguration setLogAnnotationsCountLimit(int value) {
     return this;
   }
 
+  public boolean isEnableHyperlogsConvergedCsp() {
```
Comment (joannak-vmware) [inline java-lib/src/main/java/com/wavefront/api/agent/ValidationConfiguration.java:221]: nit: can we rename this to `hyperlogsConvergedCspEnabled`?

Comment (joannak-vmware) [reply java-lib/src/main/java/com/wavefront/api/agent/ValidationConfiguration.java:221]: or `enableHyperlogsConvergedCsp`

File: java-lib/src/main/java/com/wavefront/api/agent/ValidationConfiguration.java (hunk 2 above)
Comment (deosu) [reply java-lib/src/main/java/com/wavefront/api/agent/ValidationConfiguration.java]: done

File: java-lib/src/main/java/com/wavefront/api/agent/ValidationConfiguration.java (hunk 3 above)
Comment (deosu) [reply java-lib/src/main/java/com/wavefront/api/agent/ValidationConfiguration.java]: done

**Review by amitw-vmware — DISMISSED** [review]
lgtm

**Review by joannak-vmware — APPROVED** [review]
LGTM

**Pull request merged**